File: Exercism Codes/JavaScript/triangle.js
export class Triangle {
    constructor(...sides) {
        this.s = sides;
    }
    static zeroAndInequality(s) {
        return s.every(c => c > 0) 
          && s[0] + s[1] >= s[2] 
          && s[1] + s[2] >= s[0] 
          && s[0] + s[2] >= s[1];
    }
    get isEquilateral() {
        return this.s[0] == this.s[1] 
          && this.s[1] == this.s[2] 
          && Triangle.zeroAndInequality(this.s);
    }
    get isIsosceles() {
        return (this.s[0] == this.s[1] 
          || this.s[1] == this.s[2] 
          || this.s[0] == this.s[2]) 
          && Triangle.zeroAndInequality(this.s);
    }
    get isScalene() {
        return this.s[0] != this.s[1] 
          && this.s[1] != this.s[2] 
          && this.s[0] != this.s[2] 
          && Triangle.zeroAndInequality(this.s);
    }
}

File: Exercism Codes/Kotlin/Triangle.kt
class Triangle<out T : Number>(a: T, b: T, c: T) {
    private val da = a.toDouble()
    private val db = b.toDouble()
    private val dc = c.toDouble()
    init {
        if (
            !(da + db >= dc && db + dc >= da && da + dc >= db) ||
            (da == 0.0 || db == 0.0 || dc == 0.0)
            ) {
            throw IllegalArgumentException()
        }
    }
    val isEquilateral: Boolean = da == db && db == dc
    val isIsosceles: Boolean = da == db || da == dc || db == dc
    val isScalene: Boolean = da != db && db != dc && da != dc
}